main.go

```diff
@@ -135,6 +129,6 @@ func loadControllerConfig() (config.ControllerConfig, error) {
 
 // getLoggerWithLogLevel returns logger with specific log level.
 func getLoggerWithLogLevel(logLevel string, logFilePath string) (logr.Logger, error) {
-	ctrlLogger := logger.New(logLevel, logFilePath)
+	ctrlLogger := logger.New(logLevel, logFilePath, 2)
```

Comment:
Nit: later we can move the logger defaults of 0,1,2..to utils as enum

Reply:
Yeah, I considered an enum, but could not come up with any names that make sense, as the value to use is dependent on where the logger will be called from. If you are within a goroutine function, you want 0, as above is the standard library runtime. If you are within the controller handle routines, you want 2, as you want to skip the controller call stack surrounding the function